docs/flutter_status.md

```diff
@@ -0,0 +1,98 @@
+# Current Flutter status
+
+## Firebase
+
+1. Firebase can run flutter tests. You can find example in ```test_projects/flutter```,
+   simple run ```./build_and_run_tests_firebase```.
+
+1. Firebase not supporting sharding for Flutter.
+
+## Flank
+
+Currently, Flank cannot run any flutter tests.
+That happens because  ```app-debug-androidTest.apk``` contains only the test class with the rule. All tests are in ```app-debug.apk```.
+
+## How to create flutter tests for firebase
+
+1. Android side
+
+```java
+
+@RunWith(FlutterTestRunner.class)
+public class MainActivityTest {
+    @Rule
+    public ActivityTestRule<MainActivity> rule = new ActivityTestRule<>(MainActivity.class, true, false);
+}
+
+```
+
+1. Flutter side
+
+This is the test app entry point.
+
+```dart
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  testWidgets("success test example", (WidgetTester tester) async {
+    // Wait for app widget
+    await tester.pumpWidget(MyApp());
+
+    // Verify that our counter starts at 0.
+    expect(find.text('0'), findsOneWidget);
+    expect(find.text('1'), findsNothing);
+
+    // Tap the '+' icon and trigger a frame.
+    await tester.tap(find.byIcon(Icons.add));
+    await tester.pump();
+
+    // Verify that our counter has incremented.
+    expect(find.text('0'), findsNothing);
+    expect(find.text('1'), findsOneWidget);
+  });
+}
+
+```
+
+You need to compile the Flutter app with few steps
+
+```shell
+flutter build apk
+```
+
+```shell
+./gradlew app:assembleAndroidTest
+```
+
+```shell
+./gradlew app:assembleAndroidTest
+```
+
+```shell
+./gradlew app:assembleDebug -Ptarget="path to test entry point eg. $dir/integration_tests/integration_tests.dart"
+```
+
+All dart tests are placed in ```app-debug.apk``` instead of ```app-debug-androidTest.apk```.
+That's why Flank doesn't see flutter tests.
+
+## How flutter tests work in firebase
+
+Flutter Integration Test plugin use ```channel``` to communicate between ```FlutterTestRunner```on android side
+and dart code.
+When dart tests was finished, flutter send information about test results to native code. Native code
+receive information's in ```IntegrationTestPlugin``` in method ```onMethodCall```. ```FlutterTestRunner``` sets test
+statuses in method ```run``` (lines: 78, 82, 84).
```

Comment:
Would be great to have hyperlinks for specified lines.

Reply:
Thanks! Hyperlinks added!